crates/sage/src/scoring.rs

```diff
@@ -170,6 +186,16 @@ impl Score {
     }
 }
 
+impl Score {
+    /// Calculate the hyperscore for a given PSM choosing between implementations based on `score_type`
+    fn hyperscore(&self, score_type: ScoreType) -> f64 {
+        match score_type {
```

Comment:
Match expression is unnecessary here, you can simply write `score_type.score(...)`